docs/specifications/tests/Zone-TP/zone01.md

```diff
@@ -3,43 +3,264 @@
+by the SOA. It should not contain the name of the zone itself. That information
+would be useless, as to discover it, one needs to start with the domain name of
+the SOA record - that is the name of the zone".
 
-If the SOA MNAME field is not an authoritative DNS server for the domain name,
-this test case fails.
+There exists an unstandardized practice to set the SOA MNAME to ".", which
+should not be interpreted that there is no primary master server, but to
+indicate that there is no default server for dynamic updates. With ".", SOA
+MNAME has no server name. There is at least one old and expired Internet-Draft
+that attempted to standardize that behavior, [draft-jabley-dnsop-missing-mname].
+If the SOA MNAME is an empty name (".") this Test Case will not try to connect
+to a server behind it since there will never be a server behind that name, as
+the purpose is most definitely to follow that practice. Instead, a special
+message will be outputted.
+ 
+This Test Case will check that:
+ - the SOA MNAME contains the master name server of *Child Zone*,
+   as best as it can be determined.
+ - the SOA MNAME name server is authoritative of *Child Zone*.
+ - the SOA SERIAL of the SOA MNAME is at least equal to the ones found
+   from the name servers in the NS record set of *Child Zone*.
+   This comparison must be done following [RFC1982].
+ - the SOA MNAME name server is listed as part of the NS record set of *Child Zone*.
 
-### Special procedural requirements
+## Scope
+
+It is assumed that *Child Zone* is tested and reported by [CONNECTIVITY01].
+This Test Case will just ignore non-responsive name servers or name servers
+not giving a correct DNS response for an authoritative name server, except
+if the name server is listed in the SOA MNAME.
+
+The syntax of the SOA MNAME for *Child Zone* is not tested in this Test
+Case, see [SYNTAX07].
+
+The consistency of the SOA MNAME between different servers of *Child Zone*
+is not tested by this Test Case, see [CONSISTENCY06].
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+
+Message Tag                   | Level   | Arguments             | Message ID for message tag
+:---------------------------- |:--------|:----------------------|:---------------------------------------------------------------------------------------
+Z01_MNAME_HAS_LOCALHOST_ADDR  | WARNING | nsname, ns_ip         | SOA MNAME name server "{nsname}" resolves to a localhost IP address ({ns_ip}).
+Z01_MNAME_IS_DOT              | NOTICE  | ns_ip_list            | SOA MNAME is specified as "." which usually means "no server". Fetched from name servers "{ns_ip_list}".
+Z01_MNAME_IS_LOCALHOST        | WARNING | ns_ip_list            | SOA MNAME name server is "localhost", which is invalid. Fetched from name servers "{ns_ip_list}".
+Z01_MNAME_IS_MASTER           | DEBUG   | ns                    | SOA MNAME name server "{ns}" appears to be master.
```

Comment:
I guess it should be `ns_list` since when it is outputted there could be multiple `ns`:
```
13. If the set *MNAME Is Master* is non-empty, then output *[Z01_MNAME_IS_MASTER]*
    with SOA MNAME name server name(s) and IP address(es) from the set.
```